gtest/src/log.rs

```diff
@@ -396,6 +396,28 @@ impl RunResult {
             .flat_map(DecodedCoreLog::try_from_log)
             .collect()
     }
+
+    /// If the main message panicked.
+    pub fn panicked(&self) -> bool {
+        self.log.len() == 1
+            && matches!(
+                self.log[0].reply_code(),
+                Some(ReplyCode::Error(ErrorReplyReason::Execution(
+                    SimpleExecutionError::UserspacePanic
+                )))
+            )
+    }
+
+    /// If the main message panicked with a given message.
```

Comment:
I would add that this test is the best to check the `msg` to be a substring of the error payload. So if it returns false, it doesn't necessarily mean that there wasn't panic.